R/build-home-community.R

```diff
@@ -0,0 +1,30 @@
+has_contributing <- function(path = ".") {
```

Comment:
Shouldn't you be checking for the `.md` files in `.github` instead of the HTML?

Reply:
I thought about doing that but I have a package where the COC is top-level for whatever reason (https://github.com/ropenscilabs/rodev/community ) so looking for the html seemed more general. Do you want me to change that?

Reply:
https://github.com/ropenscilabs/rodev has a top-level *md* file, not an *html* file.

Reply:
GitHub itself lets you add the code of conduct to the root or .github/ https://help.github.com/en/articles/adding-a-code-of-conduct-to-your-project, and it's the same for the contributing guidelines https://help.github.com/en/articles/setting-guidelines-for-repository-contributors. 

But if you want me to look for the .md instead in both places, I can. :wink:

Reply:
You should look in both places, yes. But the other issue is that your `has_coc` and `has_contributing` check for HTML files, not .md files.

I think you want the following:

```r

has_contributing <- function(path = ".") {
  file_exists(path(path, 'CONTRIBUTING.md')) || file_exists(path(path, '.github', 'CONTRIBUTING.md'))
}

has_coc <- function(path = ".") {
  file_exists(path(path, 'CODE_OF_CONDUCT.md')) || file_exists(path(path, '.github', 'CODE_OF_CONDUCT.md'))
}

Reply:
yes but pkgdown builds the md (no matter where it was, either in root or in .github) into an html file, and the html is always at the same place whereas the md can be in root or in .github so it'd make the code longer but I can switch. Is that your preference?

Reply:
Ah, that makes more sense. It sounds like it's more correct the way it is now..

Reply:
It's in any case _easier_ now :grin:

Reply:
It feels wrong to use information about the built site to generate the site data. It's going to be more work to check for the sources, but I think it's important, otherwise you've introduced a subtle order dependency between the different pieces.

Reply:
Okay, that makes sense, I re-wrote that!